src/services.rs

```diff
@@ -1,118 +0,0 @@
-            custom: CustomServiceServer::new(mission_rpc.clone()),
-            group: GroupServiceServer::new(mission_rpc.clone()),
-            hook: HookServiceServer::new(hook_rpc),
-            mission: MissionServiceServer::new(mission_rpc.clone()),
-            net: NetServiceServer::new(mission_rpc.clone()),
-            timer: TimerServiceServer::new(mission_rpc.clone()),
-            trigger: TriggerServiceServer::new(mission_rpc.clone()),
-            unit: UnitServiceServer::new(mission_rpc.clone()),
-            world: WorldServiceServer::new(mission_rpc),
-        }
-    }
-}
-
-impl NamedService for DcsServices {
-    const NAME: &'static str = "";
-}
-
-impl Service<http::Request<transport::Body>> for DcsServices {
-    type Response = http::Response<BoxBody>;
-    type Error = Never;
-    type Future = tonic::codegen::BoxFuture<Self::Response, Self::Error>;
-
-    #[inline]
-    fn poll_ready(
-        &mut self,
-        _: &mut std::task::Context<'_>,
-    ) -> std::task::Poll<Result<(), Self::Error>> {
-        std::task::Poll::Ready(Ok(()))
-    }
-
-    fn call(&mut self, req: tonic::codegen::http::Request<transport::Body>) -> Self::Future {
-        let path = req.uri().path().trim_start_matches('/');
-        if path.starts_with(AtmosphereServiceServer::<MissionRpc>::NAME) {
-            self.atmosphere.call(req)
-        } else if path.starts_with(CoalitionServiceServer::<MissionRpc>::NAME) {
-            self.coalition.call(req)
-        } else if path.starts_with(ControllerServiceServer::<MissionRpc>::NAME) {
-            self.controller.call(req)
-        } else if path.starts_with(CustomServiceServer::<MissionRpc>::NAME) {
-            self.custom.call(req)
-        } else if path.starts_with(GroupServiceServer::<MissionRpc>::NAME) {
-            self.group.call(req)
-        } else if path.starts_with(HookServiceServer::<HookRpc>::NAME) {
-            self.hook.call(req)
-        } else if path.starts_with(MissionServiceServer::<MissionRpc>::NAME) {
-            self.mission.call(req)
-        } else if path.starts_with(NetServiceServer::<MissionRpc>::NAME) {
-            self.net.call(req)
-        } else if path.starts_with(TimerServiceServer::<MissionRpc>::NAME) {
-            self.timer.call(req)
-        } else if path.starts_with(TriggerServiceServer::<MissionRpc>::NAME) {
-            self.trigger.call(req)
-        } else if path.starts_with(UnitServiceServer::<MissionRpc>::NAME) {
-            self.unit.call(req)
-        } else if path.starts_with(WorldServiceServer::<MissionRpc>::NAME) {
-            self.world.call(req)
-        } else {
-            Box::pin(std::future::ready(Ok(http::Response::builder()
-                .status(200)
-                .header("grpc-status", "12")
-                .header("content-type", "application/grpc")
-                .body(tonic::codegen::empty_body())
-                .unwrap())))
-        }
```

Comment:
With the dependency upgrades, this code wasn't working anymore. It was always returning the `else` path. The code was initially added due to compile time issues (in https://github.com/DCS-gRPC/rust-server/pull/61). Fortunately, the compile time issues are gone (either due to a Rust or dep update). At least for me - if you have the time, please check the compile time for you too. This is why I've removed the workaround again instead of fixing it.

Reply:
I will test this tomorrow locally

Reply:
@rurounijones I'll wait for your feedback before merging 👍

Reply:
Am receiving the following errors on this branch and this branch only. I re-ran on main branch without issues.

I chopped off only to the last error and message. Lemme know if you need the entire output


```
   C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(245,5): error MSB8066: Custom build for 'C:\Users\jeff\source\repos\DCS-gRPC\rust-server\target\debug\build\prost-build-68bc37c7b395e905\out\build\CMakeFiles\3800ac9a10098988c9e6c56711b08d23\map_lite_unittest.pb.cc.rule;C:\Users\jeff\source\repos\DCS-gRPC\rust-server\target\debug\build\prost-build-68bc37c7b395e905\out\build\CMakeFiles\3800ac9a10098988c9e6c56711b08d23\unittest_import_lite.pb.cc.rule;C:\Users\jeff\source\repos\DCS-gRPC\rust-server\target\debug\build\prost-build-68bc37c7b395e905\out\build\CMakeFiles\3800ac9a10098988c9e6c56711b08d23\unittest_import_public_lite.pb.cc.rule;C:\Users\jeff\source\repos\DCS-gRPC\rust-server\target\debug\build\prost-build-68bc37c7b395e905\out\build\CMakeFiles\3800ac9a10098988c9e6c56711b08d23\unittest_lite.pb.cc.rule;C:\Users\jeff\.cargo\registry\src\github.com-1ecc6299db9ec823\prost-build-0.10.3\third-party\protobuf\cmake\CMakeLists.txt' exited with code 3. [C:\Users\jeff\source\repos\DCS-gRPC\rust-server\target\debug\build\prost-build-68bc37c7b395e905\out\build\lite-test.vcxproj]
    Generating Code...
    test_plugin.vcxproj -> C:\Users\jeff\source\repos\DCS-gRPC\rust-server\target\debug\build\prost-build-68bc37c7b395e905\out\build\Debug\test_plugin.exe

  --- stderr
  CMake Warning:
    Manually-specified variables were not used by the project:

      CMAKE_ASM_FLAGS
      CMAKE_ASM_FLAGS_DEBUG


  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 1

  build script failed, must exit now', C:\Users\jeff\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.48\src\lib.rs:975:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Hmm, it looks like it tries to build protoc for you instead of using the bundled one. If I am reading [the code](https://github.com/tokio-rs/prost/blob/master/prost-build/build.rs#L96) correctly, it should prefer the bundled one over compiling it from source. The difference between our systems might be that I don't have `cmake` installed.

From the error I am assuming you are using Visual Studio for the build, could you try if you get the same error when simply running `cargo build` from any console? This might help to narrow the cause of the error down.

Reply:
From a powershell console. I have no idea where `"-G" "Visual Studio 17 2022` is coming from.

```
cargo build
   Compiling prost-build v0.10.3
error: failed to run custom build command for `prost-build v0.10.3`

Caused by:
  process didn't exit successfully: `C:\Users\jeff\source\repos\DCS-gRPC\rust-server\target\debug\build\prost-build-7115271569e6896d\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-changed=C:\Users\jeff\.cargo\registry\src\github.com-1ecc6299db9ec823\prost-build-0.10.3\third-party\protobuf\cmake
  CMAKE_TOOLCHAIN_FILE_x86_64-pc-windows-msvc = None
  CMAKE_TOOLCHAIN_FILE_x86_64_pc_windows_msvc = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-pc-windows-msvc = None
  CMAKE_GENERATOR_x86_64_pc_windows_msvc = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-pc-windows-msvc = None
  CMAKE_PREFIX_PATH_x86_64_pc_windows_msvc = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-pc-windows-msvc = None
  CMAKE_x86_64_pc_windows_msvc = None
  HOST_CMAKE = None
  CMAKE = None
  running: "cmake" "C:\\Users\\jeff\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\prost-build-0.10.3\\third-party\\protobuf\\cmake" "-G" "Visual Studio 17 2022" "-Thost=x64" "-Ax64" "-DCMAKE_INSTALL_PREFIX=C:\\Users\\jeff\\source\\repos\\DCS-gRPC\\rust-server\\target\\debug\\build\\prost-build-68bc37c7b395e905\\out" "-DCMAKE_C_FLAGS= -nologo -MD -Brepro" "-DCMAKE_C_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_BUILD_TYPE=Debug"

  --- stderr
  thread 'main' panicked at '
  failed to execute command: The system cannot find the file specified. (os error 2)
  is `cmake` not installed?

  build script failed, must exit now', C:\Users\jeff\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.48\src\lib.rs:975:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
After running `cargo clean` and then trying to build again, I am encountering the same error. I am a bit confused that this wasn't caught by CI. Anyway, I've pushed a ~fix~ workaround. Should work now. I btw. also have the `"-G" "Visual Studio 17 2022` in there, so probably normal on Windows.

Would you mind giving it another try, too?

Reply:
Confirmed all good